### DISCUSSION TITLE
Mes tables pour la bd

### DISCUSSION THREAD

**Discussion by meiry11** [AdrienDSlone/slowTripPlanner#39] [2. Data]
-- MySQL Workbench Forward Engineering

SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';

-- -----------------------------------------------------
-- Schema mydb
-- -----------------------------------------------------
-- -----------------------------------------------------
-- Schema StripLowPlaner
-- -----------------------------------------------------

-- -----------------------------------------------------
-- Schema StripLowPlaner
-- -----------------------------------------------------
CREATE SCHEMA IF NOT EXISTS `StripLowPlaner` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci ;
USE `StripLowPlaner` ;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Balise`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Balise` (
  `idBalise` INT NOT NULL AUTO_INCREMENT,
  `date` DATE NULL DEFAULT NULL,
  `pointArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `nomPoint` VARCHAR(45) NULL DEFAULT NULL,
  `distance` INT NULL DEFAULT NULL,
  `heureDepart` VARCHAR(45) NULL DEFAULT NULL,
  `heureArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `positionDansEtape` VARCHAR(45) NULL DEFAULT NULL,
  `description` VARCHAR(45) NULL DEFAULT NULL,
  `isVerrouillé` TINYINT NULL DEFAULT NULL,
  PRIMARY KEY (`idBalise`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Equipement`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Equipement` (
  `idEquipement` INT NOT NULL,
  `Equipementvelo` VARCHAR(45) NULL DEFAULT NULL,
  `EquipementTrot` VARCHAR(45) NULL DEFAULT NULL,
  `prix` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idEquipement`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`MoyenDeLocomotion`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`MoyenDeLocomotion` (
  `idMoyenDeLocomotion` INT NOT NULL AUTO_INCREMENT,
  `isVelo` TINYINT NULL DEFAULT NULL,
  `isTrotinette` TINYINT NULL DEFAULT NULL,
  `marque` VARCHAR(45) NULL DEFAULT NULL,
  `verificationPneus` TINYINT NULL DEFAULT NULL,
  PRIMARY KEY (`idMoyenDeLocomotion`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`NiveauDifficulté`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`NiveauDifficulté` (
  `idNiveauDifficulté` INT NOT NULL AUTO_INCREMENT,
  `faible` VARCHAR(45) NULL DEFAULT NULL,
  `moyen` VARCHAR(45) NULL DEFAULT NULL,
  `fort` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idNiveauDifficulté`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Parametres`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Parametres` (
  `idParametres` INT NOT NULL AUTO_INCREMENT,
  `kmEtapeMoyenne` INT NULL DEFAULT NULL,
  `vitesseDesirée` INT NULL DEFAULT NULL,
  `frequenceRepos` VARCHAR(45) NULL DEFAULT NULL,
  `frequencePause` VARCHAR(45) NULL DEFAULT NULL,
  `dureePause` VARCHAR(45) NULL DEFAULT NULL,
  `maxKilometrageJour` INT NULL DEFAULT NULL,
  PRIMARY KEY (`idParametres`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Parcours`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Parcours` (
  `idParcours` INT NOT NULL AUTO_INCREMENT,
  `pointDepart` VARCHAR(45) NULL DEFAULT NULL,
  `dateDepart` DATE NULL DEFAULT NULL,
  `dateFin` DATE NULL DEFAULT NULL,
  `pointArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `nomParcours` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idParcours`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Preferences`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Preferences` (
  `idPreferences` INT NOT NULL,
  PRIMARY KEY (`idPreferences`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Voyageur`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Voyageur` (
  `idVoyageur` INT NOT NULL AUTO_INCREMENT,
  `nom` VARCHAR(45) NULL DEFAULT NULL,
  `prenom` VARCHAR(45) NULL DEFAULT NULL,
  `adresse` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idVoyageur`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;

SET SQL_MODE=@OLD_SQL_MODE;
SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS;
SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS;

**Comment by AdrienDSlone** [comment]
Merci Mariam :)

J'ai reverse engeneering ton code et voici le schéma mais il manque les relations entre les tables :/
Pourrais tu les ajouter dans ton code stp ?
![image](https://user-images.githubusercontent.com/50139099/176035522-7f3511b1-f26e-4229-a445-42d9260bf72c.png)

**Comment by meiry11** [comment]
-- MySQL Workbench Forward Engineering

SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';

-- -----------------------------------------------------
-- Schema mydb
-- -----------------------------------------------------
-- -----------------------------------------------------
-- Schema StripLowPlaner
-- -----------------------------------------------------

-- -----------------------------------------------------
-- Schema StripLowPlaner
-- -----------------------------------------------------
CREATE SCHEMA IF NOT EXISTS `StripLowPlaner` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci ;
USE `StripLowPlaner` ;

-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Parcours`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Parcours` (
  `idParcours` INT NOT NULL AUTO_INCREMENT,
  `pointDepart` VARCHAR(45) NULL DEFAULT NULL,
  `dateDepart` DATE NULL DEFAULT NULL,
  `dateFin` DATE NULL DEFAULT NULL,
  `pointArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `nomParcours` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idParcours`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Balise`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Balise` (
  `idBalise` INT NOT NULL AUTO_INCREMENT,
  `date` DATE NULL DEFAULT NULL,
  `pointArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `nomPoint` VARCHAR(45) NULL DEFAULT NULL,
  `distance` INT NULL DEFAULT NULL,
  `heureDepart` VARCHAR(45) NULL DEFAULT NULL,
  `heureArrivée` VARCHAR(45) NULL DEFAULT NULL,
  `positionDansEtape` VARCHAR(45) NULL DEFAULT NULL,
  `description` VARCHAR(45) NULL DEFAULT NULL,
  `isVerrouillé` TINYINT NULL DEFAULT NULL,
  `Parcours_idParcours` INT NOT NULL,
  PRIMARY KEY (`idBalise`, `Parcours_idParcours`),
  INDEX `fk_Balise_Parcours1_idx` (`Parcours_idParcours` ASC) VISIBLE,
  CONSTRAINT `fk_Balise_Parcours1`
    FOREIGN KEY (`Parcours_idParcours`)
    REFERENCES `StripLowPlaner`.`Parcours` (`idParcours`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Equipement`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Equipement` (
  `idEquipement` INT NOT NULL,
  `Equipementvelo` VARCHAR(45) NULL DEFAULT NULL,
  `EquipementTrot` VARCHAR(45) NULL DEFAULT NULL,
  `prix` VARCHAR(45) NULL DEFAULT NULL,
  PRIMARY KEY (`idEquipement`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Voyageur`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Voyageur` (
  `idVoyageur` INT NOT NULL AUTO_INCREMENT,
  `nom` VARCHAR(45) NULL DEFAULT NULL,
  `prenom` VARCHAR(45) NULL DEFAULT NULL,
  `adresse` VARCHAR(45) NULL DEFAULT NULL,
  `Equipement_idEquipement` INT NOT NULL,
  `Parcours_idParcours` INT NOT NULL,
  PRIMARY KEY (`idVoyageur`, `Equipement_idEquipement`, `Parcours_idParcours`),
  INDEX `fk_Voyageur_Equipement1_idx` (`Equipement_idEquipement` ASC) VISIBLE,
  INDEX `fk_Voyageur_Parcours1_idx` (`Parcours_idParcours` ASC) VISIBLE,
  CONSTRAINT `fk_Voyageur_Equipement1`
    FOREIGN KEY (`Equipement_idEquipement`)
    REFERENCES `StripLowPlaner`.`Equipement` (`idEquipement`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_Voyageur_Parcours1`
    FOREIGN KEY (`Parcours_idParcours`)
    REFERENCES `StripLowPlaner`.`Parcours` (`idParcours`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`MoyenDeLocomotion`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`MoyenDeLocomotion` (
  `idMoyenDeLocomotion` INT NOT NULL AUTO_INCREMENT,
  `isVelo` TINYINT NULL DEFAULT NULL,
  `isTrotinette` TINYINT NULL DEFAULT NULL,
  `marque` VARCHAR(45) NULL DEFAULT NULL,
  `verificationPneus` TINYINT NULL DEFAULT NULL,
  `Voyageur_idVoyageur` INT NOT NULL,
  PRIMARY KEY (`idMoyenDeLocomotion`, `Voyageur_idVoyageur`),
  INDEX `fk_MoyenDeLocomotion_Voyageur_idx` (`Voyageur_idVoyageur` ASC) VISIBLE,
  CONSTRAINT `fk_MoyenDeLocomotion_Voyageur`
    FOREIGN KEY (`Voyageur_idVoyageur`)
    REFERENCES `StripLowPlaner`.`Voyageur` (`idVoyageur`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`NiveauDifficulté`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`NiveauDifficulté` (
  `idNiveauDifficulté` INT NOT NULL AUTO_INCREMENT,
  `faible` VARCHAR(45) NULL DEFAULT NULL,
  `moyen` VARCHAR(45) NULL DEFAULT NULL,
  `fort` VARCHAR(45) NULL DEFAULT NULL,
  `Parcours_idParcours` INT NOT NULL,
  PRIMARY KEY (`idNiveauDifficulté`, `Parcours_idParcours`),
  INDEX `fk_NiveauDifficulté_Parcours1_idx` (`Parcours_idParcours` ASC) VISIBLE,
  CONSTRAINT `fk_NiveauDifficulté_Parcours1`
    FOREIGN KEY (`Parcours_idParcours`)
    REFERENCES `StripLowPlaner`.`Parcours` (`idParcours`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Parametres`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Parametres` (
  `idParametres` INT NOT NULL AUTO_INCREMENT,
  `kmEtapeMoyenne` INT NULL DEFAULT NULL,
  `vitesseDesirée` INT NULL DEFAULT NULL,
  `frequenceRepos` VARCHAR(45) NULL DEFAULT NULL,
  `frequencePause` VARCHAR(45) NULL DEFAULT NULL,
  `dureePause` VARCHAR(45) NULL DEFAULT NULL,
  `maxKilometrageJour` INT NULL DEFAULT NULL,
  PRIMARY KEY (`idParametres`))
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


-- -----------------------------------------------------
-- Table `StripLowPlaner`.`Preferences`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `StripLowPlaner`.`Preferences` (
  `idPreferences` INT NOT NULL,
  `NiveauDifficulté_idNiveauDifficulté` INT NOT NULL,
  `Parametres_idParametres` INT NOT NULL,
  PRIMARY KEY (`idPreferences`, `NiveauDifficulté_idNiveauDifficulté`, `Parametres_idParametres`),
  INDEX `fk_Preferences_NiveauDifficulté1_idx` (`NiveauDifficulté_idNiveauDifficulté` ASC) VISIBLE,
  INDEX `fk_Preferences_Parametres1_idx` (`Parametres_idParametres` ASC) VISIBLE,
  CONSTRAINT `fk_Preferences_NiveauDifficulté1`
    FOREIGN KEY (`NiveauDifficulté_idNiveauDifficulté`)
    REFERENCES `StripLowPlaner`.`NiveauDifficulté` (`idNiveauDifficulté`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_Preferences_Parametres1`
    FOREIGN KEY (`Parametres_idParametres`)
    REFERENCES `StripLowPlaner`.`Parametres` (`idParametres`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB
DEFAULT CHARACTER SET = utf8mb4
COLLATE = utf8mb4_0900_ai_ci;


SET SQL_MODE=@OLD_SQL_MODE;
SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS;
SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS;

**Comment by meiry11** [comment]
![Bdd_stp](https://user-images.githubusercontent.com/73870215/176164501-de2af4ef-31c0-421d-8f7c-a25946b750c5.PNG)

**Reply by AdrienDSlone** [reply]
Pour le niveau de difficulté, je pense qu'il n'y a besoin que d'un intitulé et ce sera cet intitulé qui sera "faible, moyen, fort"

Dans moyen de locomotion, c'est très bien vu les informations pour vérifier la pression des pneus !

le voyage doit impérativement avoir un email et un mdp

Pour l'équipement, ce sera comme pour difficulté, on va ouvrir une table et y lister tout ce qu'il a en équipement. Si tu veux attribuer un équipement à moyen de locomotion, il faut une table relation n-n entre l'équipement et le moyen de locomotion qui le reçoit.

Pour les paramètres et les balises, ça rejoint ce que j'imaginais, à voir jusqu'où on pousse la personnalisation. Je sais que ça avait fait beaucoup débat.

**Reply by AdrienDSlone** [reply]
Au sujet de l'équipement, j'avais attribué l'équipement à des parcours
https://user-images.githubusercontent.com/50139099/172566691-6448c59d-e2cb-4e54-9ed7-39d5b586a330.png

et j'avais gardé ça en fusionnant avec celle de Nicolas :
https://github.com/AdrienDSlone/slowTripPlanner/discussions/34#discussion-4131592

**Reply by meiry11** [reply]
si nous pouvons nous mettre à 4 demain pour pour finaliser la bdd ça sera l'Ideal
Je pourrai commencer les insertions demain soir ou passer à l'étape suivante que nous aurons choisi

**Reply by AdrienDSlone** [reply]
Yes :)

**Comment by jdenizart** [comment]
![SlowTripPlanner - BdD - Jerome v1](https://user-images.githubusercontent.com/104148814/176169098-e2555f8c-96ab-448b-a368-bb260bb85373.png)

**Reply by AdrienDSlone** [reply]
Si je comprends bien, pour l'instant il n'y aurait pas de gestion d'équipement, et il n'y aurait pas de gestion d'une liste de moyen de locomotion détachées des préférences ni de difficulté détachée ?

Avoir la difficulté et le moyen de locomotion détachés permet de faire évoluer dans le temps, si on souhaite faire varier l'échelle de difficulté ou ajouter de nouveaux moyens de locomotion.

Egalement, l'utilisateur commence par renseigner son moyen de locomotion et sa difficulté d'où découlent des préférences automatiquement calculées mais également modifiables à la main.

Dans la version 1 du projet, je privilégierais les tables moyens de locomotion et difficulté, toutes deux reliées à la table préférences, mais on s'occuperait pas de la table préférences car ses champs impliquent d'entrer dans des détails techniques.

Je pense qu'on peut garder les préférences comme centralisateur des autres entitées impactant les préférences, par exemple, on pourrait imaginer une entitée "Santé", précisant si le cycliste est en bonne santé, cardiaque, son age etc, quiserait aussi relié aux préférences.

Les préférences centralisent mais pour être précis, il faut des entitées dédiées à ce qui ne relève pas des données de customisation pour sportif et professionnels.

**Reply by AdrienDSlone** [reply]
L'utilisateur doit également avoir un mot de passe et je ne sais plus si j'y avais pensé mais bien vu la photo.
Du coup j'ajouterai une bannière pour être raccord avec les us et coutûmes des réseaux sociaux.

**Comment by jdenizart** [comment]
![SlowTripPlanner - BdD - Jerome v2](https://user-images.githubusercontent.com/104148814/176447839-76f2042a-652f-4365-8323-33a386c37cf5.png)
